autoload/OmniSharp.vim

```diff
@@ -759,6 +768,15 @@ function! OmniSharp#Install(...) abort
   endif
 endfunction
 
+function! ValidPowerShellSettings()
+    let l:ps_policy_level = system('powershell Get-ExecutionPolicy')
+    if l:ps_policy_level[0:9] ==# "Restricted" || l:ps_policy_level[0:8] ==# "AllSigned"
```

Comment:
This is fine as is, but when using Vimscript is is much more common to use regular expressions over substrings.

```vim
if l:ps_policy_level =~# '\(Restricted\|AllSigned\)'
```

You may also want to add the `^` anchor before the first `\` character in the above regex, to make it faster.

Reply:
Yeah, @nickspoons had brought up this, and some other vimscript concepts, in the issue thread. It is much cleaner than the substring!

Reply:
Nice refactor! I changed the way the script outputs and so instead of checking for 'Failure' it is checking for '0' as success. But, because the output includes the new line '^@' char I am still checking it like a string. If you know of a cleaner way for checking the 0 or 1 outputs  in vimscript let me know.

Reply:
Great work @atomsand1101its.

Now onto your question about checking the exit codes.

After running the `system()` function you should be able to check the `v:shell_error` variable to get the exit code.

However this may not work, because we are calling `powershell` in the `system()` function, so we might just get the exit code of PowerShell and not the script. I am also not sure if this will work as expected on Windows, so It'll need to be tested.

For more info see:
```vim
:help system()
:help v:shell_error
```

Reply:
Ha ha, we're obviously on the same page at eh moment @axvr. Yeah I tested `v:shell_error` and it works fine with the powershell script.